lib/src/multires_stochastic_texture_synthesis.rs

```diff
@@ -1191,6 +1201,213 @@ impl PrerenderedU8Function {
+            ChunkSearchInfo {
+                x: chunk_x + 1,
+                y: chunk_y,
+                center: false,
+                closest_point_on_boundary_x: (chunk_x + 1) * self.chunk_size as i32 - self.offset_x,
+                closest_point_on_boundary_y: y as i32,
+            },
+            ChunkSearchInfo {
+                x: chunk_x - 1,
+                y: chunk_y,
+                center: false,
+                closest_point_on_boundary_x: chunk_x * self.chunk_size as i32 - self.offset_x,
+                closest_point_on_boundary_y: y as i32,
+            },
+            ChunkSearchInfo {
+                x: chunk_x,
+                y: chunk_y - 1,
+                center: false,
+                closest_point_on_boundary_x: x as i32,
+                closest_point_on_boundary_y: chunk_y * self.chunk_size as i32 - self.offset_y,
+            },
+            ChunkSearchInfo {
+                x: chunk_x,
+                y: chunk_y + 1,
+                center: false,
+                closest_point_on_boundary_x: x as i32,
+                closest_point_on_boundary_y: (chunk_y + 1) * self.chunk_size as i32 - self.offset_y,
+            },
+            ChunkSearchInfo {
+                x: chunk_x + 1,
+                y: chunk_y + 1,
+                center: false,
+                closest_point_on_boundary_x: (chunk_x + 1) * self.chunk_size as i32 - self.offset_x,
+                closest_point_on_boundary_y: (chunk_y + 1) * self.chunk_size as i32 - self.offset_y,
+            },
+            ChunkSearchInfo {
+                x: chunk_x - 1,
+                y: chunk_y + 1,
+                center: false,
+                closest_point_on_boundary_x: chunk_x * self.chunk_size as i32 - self.offset_x,
+                closest_point_on_boundary_y: (chunk_y + 1) * self.chunk_size as i32 - self.offset_y,
+            },
+            ChunkSearchInfo {
+                x: chunk_x + 1,
+                y: chunk_y - 1,
+                center: false,
+                closest_point_on_boundary_x: (chunk_x + 1) * self.chunk_size as i32 - self.offset_x,
+                closest_point_on_boundary_y: chunk_y * self.chunk_size as i32 - self.offset_y,
+            },
+            ChunkSearchInfo {
+                x: chunk_x - 1,
+                y: chunk_y - 1,
+                center: false,
+                closest_point_on_boundary_x: chunk_x * self.chunk_size as i32 - self.offset_x,
+                closest_point_on_boundary_y: chunk_y * self.chunk_size as i32 - self.offset_y,
+            },
+        ];
+        // Note locking all of them at different times is fine as opposed to
+        // all at once but this behavior should be noted
+        let mut tmp_result: Vec<(i32, i32, i32)> = Vec::with_capacity(k * 9);
+        result.clear();
+        result.reserve(k);
+
+        // this isn't really the kth best distance but should be good enough for how simple it is
+        let mut kth_best_squared_distance = i32::max_value();
```

Comment:
Using `i32` here starts feeling a bit tight -- some people use _giant_ images, and there's a chance of this overflowing, particularly in the first stage of processing before "fanning out" to a larger grid of trees.

Reply:
oh wow never considered an image of that size. I'll bump to i64